arbitrum/multigas/constraint_test.go

```diff
@@ -0,0 +1,53 @@
+package multigas
+
+import (
+	"testing"
+	"time"
+)
+
+func assertEqual[T comparable](t *testing.T, description string, got T, want T) {
+	t.Helper()
```

Comment:
Optional:

This is definitely a matter of preference, but I don't like helper functions like this one. I think it makes it harder to follow the flow of the places where the tests could have failed.

Also, I don't love that it always uses `t.Fatalf` when I believe there are places in the test where `t.Errorf` would make more sense. For example. Sure, fail the test if the wrong number of constraints were created because later tests aren't going to be able to run. But, when checking of a particular constraint has the right value, we should probably get errors for each different constraint that has the wrong value, and `t.Errorf` would give us that.